### PR TITLE
Transport EAFP

### PR DESCRIPTION
**Pull request by jlee-made**
Don't try to detect if transport is connected, just try it and reconnect once if it fails

### CODE REVIEW

File: src/striemann/metrics.py [hunk 1]
```diff
@@ -111,37 +113,43 @@ def __init__(self, host="localhost", port="5555", timeout=5):
...
 
     def send_event(self, event):
         riemann_event = Client.create_event(event)
         self._message.events.add().MergeFrom(riemann_event)
 
     def _ensure_connected(self):
-        if not self.is_connected():
+        # This is just to avoid logging about failure on the first try
+        if not self._connected:
+            self._connected = True
```
Comment (bobthemighty) [inline src/striemann/metrics.py]: Why set `_connected = True` and *then* connect? Seems backward.

File: src/striemann/metrics.py [hunk 2]
```diff
@@ -111,37 +113,43 @@ def __init__(self, host="localhost", port="5555", timeout=5):
...
+                    "Failed twice to flush metrics to riemann", exc_info=True
+                )
         if is_closing:
             self.transport.disconnect()
         self._new_message()
 
     def _new_message(self):
         self._message = Msg()
 
+    @deprecated("Should be no need to check, will reconnect automatically")
```
Comment (bobthemighty) [inline src/striemann/metrics.py]: I think I'd sooner just remove the method. We're not at a 1.0 yet, so I'm happy to make breaking changes.

File: src/striemann/tests/test_striemann.py [hunk 3]
```diff
@@ -97,3 +97,61 @@ def test_timers(self, timer):
...
+            nonlocal failed
+            if not failed:
+                failed = True
+                raise Exception('bother')
+
+        log = []
+        t.transport = FakeRiemannClientTransport(log, fail_once)
+        t.flush(is_closing=False)
+        assert log == [
+            'connected',
```
Comment (bobthemighty) [inline src/striemann/tests/test_striemann.py]: neat.

File: src/striemann/metrics.py (hunk 1 above)
Comment (jlee-made) [reply src/striemann/metrics.py]: I guess it doesn't matter much given the purpose of that (see comment)

But I'll switch it around

File: src/striemann/metrics.py (hunk 2 above)
Comment (jlee-made) [reply src/striemann/metrics.py]: Personally I never pay attention to whether the version number is less than one.  Lots of stuff never goes above 1 but is fine.  And I appreciate projects that give me a chance to fix changes before breaking APIs completely.  I think if you want to change APIs you should say "we're going to change APIs", prominently, in the README

You really never know who's using it with these things

Comment (jlee-made) [reply src/striemann/metrics.py]: of course, can remove it later

**Pull request merged**